server/src/main/java/org/elasticsearch/action/admin/indices/get/GetIndexResponse.java

```diff
@@ -315,9 +314,16 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
 
     private static ImmutableOpenMap<String, MappingMetaData> parseMappings(XContentParser parser) throws IOException {
         ImmutableOpenMap.Builder<String, MappingMetaData> indexMappings = ImmutableOpenMap.builder();
-        Map<String, Object> map = parser.map();
-        if (map.isEmpty() == false) {
-            indexMappings.put(MapperService.SINGLE_MAPPING_NAME, new MappingMetaData(MapperService.SINGLE_MAPPING_NAME, map));
+        // We start at START_OBJECT since parseIndexEntry ensures that
+        while (parser.nextToken() != Token.END_OBJECT) {
+            ensureExpectedToken(Token.FIELD_NAME, parser.currentToken(), parser::getTokenLocation);
+            parser.nextToken();
+            if (parser.currentToken() == Token.START_OBJECT) {
+                String mappingType = parser.currentName();
+                indexMappings.put(mappingType, new MappingMetaData(mappingType, parser.map()));
+            } else if (parser.currentToken() == Token.START_ARRAY) {
+                parser.skipChildren();
+            }
```

Comment:
This changes the parsing of the server-side response back to accept the "old" response format that is including types. I did this because the "new" format is now parsed in the client-side GetIndexResponse, while this parsing code should only be used by the HLRC deprecated client code that uses this GetIndexResponse, and that forces "include_type_name=true" all the time. On the server side we are only using this parsing code in round-trip tests (GetIndexResponseTests)

Reply:
Should we also undo the changes to `toXContent`, so that we always include the type name? I no longer think we need to support both the `include_type_name = true` and `include_type_name = false` paths. I think this will make reasoning about these classes quite a bit easier.

Reply:
I think we do, wasn't this the whole point of leaving the user the option to set this parameter to switch back to the new behaviour in 7.0? Leaving the HLRC aside, any user of the REST API should still be able to ask for both response versions, which will both be rendered by this client side response class.

Reply:
I'm sorry, you're completely right! I got confused because I have been staring at the 'put' side of things, and forgot that these classes actually render the REST responses.